src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/18.diff.sql

```diff
@@ -0,0 +1,297 @@
+    CREATE PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp 
+        AS PARTITION PartitionFunction_ResourceChangeData_Timestamp ALL TO([PRIMARY]);
+END; 
+
+IF (EXISTS(SELECT * FROM sys.partition_functions WHERE  name = 'PartitionFunction_ResourceChangeData_Timestamp')
+	AND EXISTS(SELECT * FROM sys.partition_schemes WHERE name = 'PartitionScheme_ResourceChangeData_Timestamp'))
+BEGIN
+   -- Creates initial partitions based on default 48-hour retention period.
+    DECLARE @numberOfPartitions int = 48;
+    DECLARE @rightPartitionBoundary datetime2(7);
+    DECLARE @currentDateTime datetime2(7) = sysutcdatetime();
+		
+	-- There will be 51 partition boundaries and 52 partitions, 48 partitions for history, 
+    -- one for the current hour, one for the next hour, and 2 partitions for start and end.
+    WHILE @numberOfPartitions >= -1 
+    BEGIN        
+        -- Rounds the start datetime to the hour.
+        SET @rightPartitionBoundary = DATEADD(hour, DATEDIFF(hour, 0, @currentDateTime) - @numberOfPartitions, 0);
+
+		-- Checks if a partition exists.
+		IF NOT EXISTS (SELECT 1 value FROM sys.partition_range_values AS prv
+						JOIN sys.partition_functions AS pf
+							ON pf.function_id = prv.function_id
+					WHERE pf.name = N'PartitionFunction_ResourceChangeData_Timestamp'
+						and CONVERT(datetime2(7), prv.value, 126) = @rightPartitionBoundary) 
+		BEGIN
+			-- Creates new empty partition by creating new boundary value and specifying NEXT USED file group.
+			ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+			ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary); 
+		END;            
+            
+        SET @numberOfPartitions -= 1;
+    END;
+END;
+
+-- Creates a staging table 
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeDataStaging')
+BEGIN
+    -- Staging table that will be used for partition switch out. 
+    CREATE TABLE dbo.ResourceChangeDataStaging 
+    (
+        Id bigint IDENTITY(1,1) NOT NULL,
+        Timestamp datetime2(7) NOT NULL CONSTRAINT DF_ResourceChangeDataStaging_Timestamp DEFAULT sysutcdatetime(),
+        ResourceId varchar(64) NOT NULL,
+        ResourceTypeId smallint NOT NULL,
+        ResourceVersion int NOT NULL,
+        ResourceChangeTypeId tinyint NOT NULL,
+        CONSTRAINT PK_ResourceChangeDataStaging_TimestampId PRIMARY KEY (Timestamp, Id)
+    ) ON [PRIMARY]
+END;
+GO
+
+-- Creates check for a partition check.
+IF NOT EXISTS(SELECT 1 FROM sys.check_constraints WHERE name = 'chk_ResourceChangeDataStaging_partition') 
+BEGIN    
+    ALTER TABLE dbo.ResourceChangeDataStaging WITH CHECK 
+        ADD CONSTRAINT chk_ResourceChangeDataStaging_partition CHECK(Timestamp < CONVERT(DATETIME2(7), '9999-12-31 23:59:59.9999999'));
+
+    ALTER TABLE dbo.ResourceChangeDataStaging CHECK CONSTRAINT chk_ResourceChangeDataStaging_partition;
```

Comment:
Is it possible that if script execution failed at line number 78 and during retry line 78 will never be executed since if condition at line 73 is evaluated to false?